NConfiguration/Serialization/SimpleTypes/Parsing/TimeSpanParser.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace NConfiguration.Serialization.SimpleTypes.Parsing
+{
+    public class TimeSpanParser
+    {
+        enum GroupName
+        {
+            Minus,
+            Day,
+            Hour,
+            Minute,
+            Second
+        }
+
+        private const RegexOptions RegularExpressionOptions =
+            RegexOptions.IgnoreCase | RegexOptions.CultureInvariant | RegexOptions.Compiled;
+        private const string MinusPattern = "(^-)";
+        private const string PatternBase = @"(\d+(?:\.\d+)?)";
+
+        private static readonly KeyValuePair<GroupName, Func<double, TimeSpan>>[] _groupNumberToSpans =
+        {
+            new KeyValuePair<GroupName, Func<double, TimeSpan>>(GroupName.Day, TimeSpan.FromDays),
+            new KeyValuePair<GroupName, Func<double, TimeSpan>>(GroupName.Hour, TimeSpan.FromHours),
+            new KeyValuePair<GroupName, Func<double, TimeSpan>>(GroupName.Minute, TimeSpan.FromMinutes),
+            new KeyValuePair<GroupName, Func<double, TimeSpan>>(GroupName.Second, TimeSpan.FromSeconds)
+        };
+
+        private static string _buildGroupRegex(GroupName groupName, string postfix, string pattern = PatternBase)
```

Comment:
Underscore `_` prefixes are only used for private _fields_ in this solution.
Rename `_buildGroupRegex` -> `BuildGroupRegex`.